### PR TITLE
Test Linux JDK 17 in CI

### PR DESCRIPTION
**Pull request by npotluri-rh**
Adding JDK version 17 with specific jenkins version to test if it runs on CI with no issues

- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [x] Link to relevant issues in GitHub or Jira (none)
- [x] Link to relevant pull requests, esp. upstream and downstream changes (none)
- [x] Ensure you have provided tests - that demonstrates feature works or fixes the issue

### CODE REVIEW

File: Jenkinsfile
Comment (MarkEWaite) [inline Jenkinsfile]: The plugin requires Jenkins 2.361.4 or newer.  Testing with Jenkins 2.356 won't work.
```suggestion
    [platform: 'linux',   jdk: '11'], // Linux first for coverage report on ci.jenkins.io
```

**Review by MarkEWaite — CHANGES_REQUESTED** [review]
One of the changes won't pass when run on ci.jenkins.io.  We can't test with a Jenkins version less than 2.361.4 because that is the minimum Jenkins version supported by this plugin.

File: Jenkinsfile
Comment (npotluri-rh) [reply Jenkinsfile]: Why did ci build not run for jdk 17?

Comment (MarkEWaite) [reply Jenkinsfile]: Because the `Jenkinsfile` is a trusted file and changes to it are only executed if you have merge permission to the repository.

Imagine the chaos that could be created by a malicious pull request submitter if their changes to Jenkinsfile were executed without review by a trusted maintainer.

**Comment by MarkEWaite** [comment]
I don't understand the motivation for the change.  JDK 17 is already being tested on Windows.  We intentionally chose to keep CI costs down.  Is there a compelling reason that we need Java 17 testing on both Windows and Linux?

File: Jenkinsfile
Comment (npotluri-rh) [reply Jenkinsfile]: I am very new to contributing to open source....In that case the PR I opened should not even trigger a CI build unless a trusted maintainer approved it...The change I made includes linux  - jdk 11, 17 and Windows...I see pipeline run for Linux -jdk 11 and windows but not linux - jdk 17 ...Exclusions/inclusions are controlled in some other place?

Comment (npotluri-rh) [reply Jenkinsfile]: For sure...Thank you for explaining things.

Comment (MarkEWaite) [reply Jenkinsfile]: No, the CI build is triggered, but your proposed change to the Jenkinsfile is ignored because you're not a maintainer.

I consider it overkill to test Java 17 on Linux and Java 17 on Windows.  If you are persuaded that you really need Linux with Java 17, then switch Windows to test with Java 11.  That way we're only running one Windows and one Linux, with one Java 11 and one Java 17.

**Comment by olivergondza** [comment]
@MarkEWaite, We are primarily interested in linux support with java 11 and 17, no objection to test windows as well. Would you say testing win+j17 provides good enough guarantee java 17 is covered on linux as well?

One thing that comes to mind is: https://github.com/jenkinsci/cloud-stats-plugin/blob/9294e95bce45bfa0779c94a824d79f91162df894/src/test/java/org/jenkinsci/plugins/cloudstats/CloudStatisticsTest.java#L209

**Comment by MarkEWaite** [comment]
> @MarkEWaite, We are primarily interested in linux support with java 11 and 17, no objection to test windows as well. Would you say testing win+j17 provides good enough guarantee java 17 is covered on linux as well?
> 
> One thing that comes to mind is:
> 
> https://github.com/jenkinsci/cloud-stats-plugin/blob/9294e95bce45bfa0779c94a824d79f91162df894/src/test/java/org/jenkinsci/plugins/cloudstats/CloudStatisticsTest.java#L209

I don't recall ever detecting a problem on Java 17 Linux that was not also detected on Java 17 Windows, but I also understand your strong desire to be absolutely certain that the code is well behaved on Java 17 Linux.  I have felt that same strong desire with a few of the plugins that I maintain.  You and @npotluri-rh have my full support for the addition of Java 17 Linux as a test configuration.

Thanks for explaining!

**Review by MarkEWaite — APPROVED** [review]

**Pull request merged**
